### ISSUE TITLE
tpu: tpu-v3-128-euw4a-51; run: shawn-bigrun44-danbooru256-clgan5_0-flood; description: BigGAN 256x256 on danbooru2019-s with CLGAN c_loss multiplier 5.0, g_flood 0.05, d_flood 0.20; logdir: gs://darnbooru-euw4a/runs/bigrun44/

### ISSUE THREAD

**Issue by shawwn** [tensorfork/tensorfork#6]
```gin
options.description = "BigGAN 256x256 on danbooru2019-s with CLGAN c_loss multiplier 5.0, g_flood 0.05, d_flood 0.20"

dataset.name = "images_256"
options.datasets = "gs://darnbooru-euw4a/datasets/danbooru2019-s/danbooru2019-s-0*"
options.random_labels = True
options.num_classes = 1000
train_imagenet_transform.crop_method = "random"
options.z_dim = 140
resnet_biggan.Generator.blocks_with_attention = "B5"
resnet_biggan.Discriminator.blocks_with_attention = "B2"

options.architecture = "resnet_biggan_arch"
ModularGAN.conditional = True
options.batch_size = 2048
options.gan_class = @ModularGAN
options.lamba = 1
options.training_steps = 250000
weights.initializer = "orthogonal"
spectral_norm.singular_value = "auto"

# Generator
G.batch_norm_fn = @conditional_batch_norm
G.spectral_norm = True
ModularGAN.g_use_ema = True
resnet_biggan.Generator.hierarchical_z = True
resnet_biggan.Generator.embed_y = True
standardize_batch.decay = 0.9
standardize_batch.epsilon = 1e-5
standardize_batch.use_moving_averages = False
standardize_batch.use_cross_replica_mean = True

# Discriminator
options.disc_iters = 2
D.spectral_norm = True
resnet_biggan.Discriminator.project_y = True

# Loss and optimizer
loss.fn = @hinge
penalty.fn = @no_penalty
ModularGAN.g_lr = 0.0000666
ModularGAN.d_lr = 0.0005
ModularGAN.g_lr_mul = 1.0
ModularGAN.d_lr_mul = 1.0
ModularGAN.g_optimizer_fn = @tf.train.AdamOptimizer
ModularGAN.d_optimizer_fn = @tf.train.AdamOptimizer
tf.train.AdamOptimizer.beta1 = 0.0
tf.train.AdamOptimizer.beta2 = 0.999

z.distribution_fn = @tf.random.normal
eval_z.distribution_fn = @tf.random.normal

run_config.iterations_per_loop = 250
run_config.save_checkpoints_steps = 250

#resnet_biggan.Generator.channel_multipliers = "16, 16, 8, 8, 4, 2, 1"
#resnet_biggan.Discriminator.channel_multipliers = "1, 2, 4, 8, 8, 16, 16"
```

Use flood loss
```gin
options.g_flood = 0.05
options.d_flood = 0.20
```

Use SimCLR with multiplier 5.0
```gin
options.gan_class = @CLGAN
CLGAN.weight_contrastive_loss_d = 5.0
```

Configure tensorboard image grid
```gin
options.image_grid_width = 4
options.image_grid_height = 4
```

**Comment by shawwn** [comment]
Mix in anime portraits, e621-s, and e621 portraits
```gin
options.datasets = "gs://darnbooru-euw4a/datasets/danbooru2019-s/danbooru2019-s-0*,gs://darnbooru-euw4a/datasets/e621-s/e621-s-0*,gs://darnbooru-euw4a/datasets/portraits/portraits-0*,gs://darnbooru-euw4a/datasets/e621-portraits-s-512/e621-portraits-s-512-0*"
```

**Comment by shawwn** [comment]
Increase generator LR by 3x
```gin
ModularGAN.g_lr_mul = 3.0
ModularGAN.d_lr_mul = 1.0
```

**Comment by shawwn** [comment]
Set SimCLR multiplier to 1.0
```gin
CLGAN.weight_contrastive_loss_d = 1.0
```

**Comment by shawwn** [comment]
Test that the LR can be adjusted dynamically.
```gin
ModularGAN.g_lr_mul = 1.001
ModularGAN.d_lr_mul = 3.001
options.g_flood = 0.05001
options.d_flood = 0.20001
```

**Comment by shawwn** [comment]
Confirmed.
![image](https://user-images.githubusercontent.com/59632/80663545-25d23400-8a49-11ea-97b0-442e1f37b573.png)

Revert changes:
```gin
ModularGAN.g_lr_mul = 1.0
ModularGAN.d_lr_mul = 3.0
options.g_flood = 0.05
options.d_flood = 0.20
CLGAN.weight_contrastive_loss_d = 1.0
```
